missioncontrol/etl/measure.py

```diff
@@ -93,18 +95,26 @@ def update_measure(platform_name, channel_name, measure_name):
         if buildstamp < window_start - channel['update_interval']:
             continue
         if not data.get(build_id):
-            # if we are on beta, try to get a more precise version (the version
-            # submitted to telemetry does not incorporate the beta number)
-            if channel_name == 'beta':
+            data[build_id] = {'version': version, 'data': []}
+        # if we are on beta, try to get a more precise version if we don't have
+        # it already (the version submitted to telemetry does not incorporate
+        # the beta number)
+        if channel_name == 'beta':
+            if (version not in beta_versions_processed and
+               not re.search('b[0-9]+$', version) and
+               data[build_id]['version'] == version):
```

Comment:
Not sure I understand why you have this last condition. Does it mean that we could have different versions associated to the same buildid?

Reply:
So it's been a while now since I wrote this (darn PTO) but I believe I added the last condition to short-circuit the search for an updated version when it is already different from what is in the presto table (and thus we've already done the update).

Maybe there is a clearer way to express this?